tools/test.py

```diff
@@ -138,6 +138,14 @@ def RunSingle(self, parallel, thread_id):
       try:
         start = datetime.now()
         output = case.Run()
+        # SmartOS has a bug that causes unexpected ECONNREFUSED errors.
+        # See https://smartos.org/bugview/OS-2767
+        # If ECONNREFUSED on SmartOS, retry the test one time.
+        if (output.UnexpectedOutput() and
+          sys.platform == 'sunos5' and 
+          'ECONNREFUSED' in output.output.stderr):
+            sys.stderr.write('ECONNREFUSED encountered on SmartOS; retrying test.')
```

Comment:
Cool, I'm happy to take it out. 

FWIW, when a test errors out, I think it just [spews the test output to stderr](https://github.com/Trott/io.js/blob/retry-smartos/tools/test.py#L192-L194), so I'm not sure we need to be that careful.  But this can totally be a discussion in a separate PR.